### DISCUSSION TITLE
assertEq returns "invalid data" when left and right values are the same

### DISCUSSION THREAD

**Discussion by MGM103** [Cyfrin/foundry-full-course-cu#1653] [Q&A] [answered]
Hi Everyone,

I am currently completing the testing in the upgradeable smart contract portion of the Advanced foundry course. I am unsure why but when I used `assertEq(left, right)` I kept getting an error: "invalid data". However, when I used: `assert(left == right)` the tests passed. Here is the code I was using:

```solidity
function testUpgrade() public {
        uint8 expectedVersion = 2;
        uint256 expectedNumber = 3;

        BoxV2 boxV2 = new BoxV2();

        upgrader.upgradeBox(proxy, address(boxV2));
        uint8 currentVersion = BoxV2(proxy).getVersion();
        assertEq(currentVersion, expectedVersion);

        BoxV2(proxy).setNumber(3);
        assertEq(BoxV2(proxy).getNumber(), expectedNumber);
    }
```

Link to my repo: https://github.com/MGM103/upgradeability-solidity

Appreciate the explanation in advance!

**Comment by EngrPips** [comment, marked as answer]
Hello, @MGM103. That seems weird, but I suggest you run `foundry` and retry. Let us know if it works after running `foundryUp`

**Reply by MGM103** [reply]
It was a weird one, but running `foundryup` did the trick and my `assertEq`'s are working again!

**Reply by EngrPips** [reply]
I am glad that worked; Foundry sometimes misbehaves when you need to install the latest version.

**Reply by PhasedNorth** [reply]
Hey! Thank you this solved the problem I was having too

**Reply by EngrPips** [reply]
Awesome, I am glad it helped you. Keep crushing it!
